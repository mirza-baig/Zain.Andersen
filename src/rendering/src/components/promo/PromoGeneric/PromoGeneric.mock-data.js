export const rightImage = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'PromoGeneric',
  },
  fields: {
    bottomCaptionDescription: {
      value:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit.',
    },
    bottomCaptionHeadline: {
      value: 'BOTTOM HEADLINE',
    },
    headlineText: {
      value: 'Generic Promo Headline Text',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eyebrowText: {
      value: 'This is a subheadline',
    },
    eyebrowLevel: {
      id: '7f10288a-beb1-4d95-a139-32df40094441',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3',
      name: 'Heading 3',
      displayName: 'Heading 3',
      fields: {
        Value: {
          value: 'h3',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    body: {
      value:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit lectus, et sagittis nibh risus vel metus. Quisque eu ornare ante, et gravida mauris.',
    },
    cta1Link: {
      value: {
        href: '/',
        text: 'Learn More',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '_blank',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta1Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'primary',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    primarImageMobile: {
      src: 'https://source.unsplash.com/random/600x400?small',
      alt: 'Unsplash Source',
      width: '600',
      height: '400',
    },
    primaryImageMobileFocusArea: {
      id: 'e6b36945-ebc0-4af8-ad95-d8268b56b7d0',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Focus/Center',
      name: 'Center',
      displayName: 'Center',
      fields: {
        Value: {
          value: 'center',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    primaryImage: {
      value: {
        src: 'https://source.unsplash.com/random/900x600?promotion',
        alt: 'Unsplash Source',
        width: '900',
        height: '600',
      },
    },
    primaryImageCaption: {
      value: 'sample caption goes here.',
    },
    imgPosition: {
      id: '870a817e-d7b9-41c7-a4c3-5c0008f2b2c1',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Position/Left',
      name: 'Right',
      displayName: 'Right',
      fields: {
        Value: {
          value: 'right',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
  },
};

export const leftImage = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'PromoGeneric',
  },
  fields: {
    bottomCaptionDescription: {
      value:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit.',
    },
    bottomCaptionHeadline: {
      value: 'BOTTOM HEADLINE',
    },
    headlineText: {
      value: 'Generic Promo Headline Text',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    eyebrowText: {
      value: 'This is a subheadline',
    },
    eyebrowLevel: {
      id: '7f10288a-beb1-4d95-a139-32df40094441',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3',
      name: 'Heading 3',
      displayName: 'Heading 3',
      fields: {
        Value: {
          value: 'h3',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    body: {
      value:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit lectus, et sagittis nibh risus vel metus. Quisque eu ornare ante, et gravida mauris.',
    },
    cta1Link: {
      value: {
        href: '/',
        text: 'Learn More',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '_blank',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta1Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: {
        Value: {
          value: 'primary',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: {
        Value: {
          value: 'arrow',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    primarImageMobile: {
      src: 'https://source.unsplash.com/random/600x400?small',
      alt: 'Unsplash Source',
      width: '600',
      height: '400',
    },
    primaryImageMobileFocusArea: {
      id: 'e6b36945-ebc0-4af8-ad95-d8268b56b7d0',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Focus/Center',
      name: 'Center',
      displayName: 'Center',
      fields: {
        Value: {
          value: 'center',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    primaryImage: {
      value: {
        src: 'https://source.unsplash.com/random/900x600?promotion',
        alt: 'Unsplash Source',
        width: '900',
        height: '600',
      },
    },
    primaryImageCaption: {
      value: 'sample caption goes here.',
    },
    imgPosition: {
      id: '870a817e-d7b9-41c7-a4c3-5c0008f2b2c1',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Position/Left',
      name: 'Left',
      displayName: 'Left',
      fields: {
        Value: {
          value: 'left',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    imageRatio: null,
  },
};
