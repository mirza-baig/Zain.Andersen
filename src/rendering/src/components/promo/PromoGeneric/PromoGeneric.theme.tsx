import { ThemeFile } from 'lib/context/ThemeContext';
import { getImagePosition } from 'lib/utils/getImagePosition';
import { Item } from '@sitecore-jss/sitecore-jss-nextjs';
import { getEnum } from 'lib/utils';
import classNames from 'classnames';

const getPosition = (pos: Item): string => {
  const imagePosition = getImagePosition('right', pos);
  return imagePosition;
};

const getImageDivCss = (imgPos: string, imgSpan: number): string => {
  const txtSpan = 12 - imgSpan;
  const startCol = imgPos === 'right' ? txtSpan + 1 : 1;
  const imageOrder = imgPos === 'right' ? 'order-first md:order-last' : 'order-first';
  const imageCss = `${imageOrder} col-span-6 cols-start-1 md:col-start-${startCol} md:col-span-${imgSpan}`;
  return imageCss;
};

const getTxtDivCss = (imgPos: string, imgSpan: number): string => {
  const txtSpan = 12 - imgSpan;
  const startSpan = imgPos === 'right' ? 1 : imgSpan + 1;
  const txtOrder = imgPos === 'right' ? 'order-last md:order-first' : 'order-last';
  const txtCss = `${txtOrder} flex items-center pt-2 col-span-6 my-auto md:col-start-${startSpan} md:col-span-${txtSpan}`;
  return txtCss;
};

const getImageInfo = (imageRatio: Item): { span: number; aspect: string } => {
  const ratio = getEnum<string>(imageRatio) ?? '4:3';

  let info = { span: 7, aspect: 'picture' };
  switch (ratio) {
    case '4:3':
      info = { span: 7, aspect: 'picture' };
      break;
    case '16:9':
      info = { span: 8, aspect: 'video' };
      break;
    case '1:1':
      info = { span: 6, aspect: 'square' };
      break;
    case '3:2':
      info = { span: 6, aspect: 'snapshot' };
      break;
    case '2:3':
      info = { span: 6, aspect: 'portrait' };
      break;
    default:
      info = { span: 6, aspect: 'square' };
      break;
  }

  return info;
};

export const PromoGenericTheme = (
  imagePosition: Item,
  imageItem: Item,
  legalFont: boolean
): ThemeFile => {
  const imgPos = getPosition(imagePosition);
  const imgInfo = getImageInfo(imageItem);

  return {
    aw: {
      classes: {
        wrapperClass: 'grid grid-cols-2 gap-s px-m md:grid-cols-12 md:gap-s promo-generic-wrapper',
        txtDivClass: classNames(getTxtDivCss(imgPos, imgInfo.span), `h-full max-w-[480px]`),
        imageDivClass: classNames(getImageDivCss(imgPos, imgInfo.span), `h-full`),
        headlineClass: 'text-sm-m lg:text-m font-heavy mb-4',
        eyebrowClass: 'text-sm-xxs lg:text-xxs text-dark-gray uppercase font-extralight mb-4',
        bodyClass: 'text-body text-dark-gray mb-4',
        buttonClasses: {
          wrapper: 'mb-m',
          cta1Classes: '',
        },
        bottomHeadingClass: '!font-sans text-sm-xxs lg:text-xxs font-bold mb-2 uppercase',
        bottomDescriptionClass: classNames(
          legalFont ? 'text-legal' : 'text-body',
          'text-dark-gray mb-4'
        ),
        imageClasses: `aspect-${imgInfo.aspect}`,
        captionClass: 'mt-xxs italic text-left mb-s',
        priceLevelWrapper: 'py-xxxs',
        priceTextClasses: 'text-gray font-sans text-xxs font-medium',
        priceLevelClasses: 'text-black font-heavy',
      },
    },
    rba: {
      classes: {
        wrapperClass: 'grid grid-cols-2 gap-s px-m md:grid-cols-12 md:gap-s promo-generic-wrapper',
        txtDivClass: classNames(getTxtDivCss(imgPos, imgInfo.span), `h-full border-t border-gray`),
        imageDivClass: classNames(getImageDivCss(imgPos, imgInfo.span), `h-full`),
        headlineClass: 'text-sm-m lg:text-l font-extra-light mb-4',
        eyebrowClass: 'text-sm-xxs lg:text-xxs text-black uppercase font-extralight mb-4',
        bodyClass: 'text-body text-dark-gray mb-4',
        buttonClasses: {
          wrapper: 'mb-m',
          cta1Classes: '',
        },
        bottomHeadingClass: 'text-sm-xxs lg:text-xxs font-heavy mb-2',
        bottomDescriptionClass: classNames(
          legalFont ? 'text-legal' : 'text-xxs',
          'mb-4 text-dark-gray'
        ),
        imageClasses: `aspect-${imgInfo.aspect}`,
        captionClass: 'mt-xxs md:mt-xxxs italic border-primary border-l-2 pl-xxs text-left mb-s',
        priceLevelWrapper: 'py-xxxs',
        priceTextClasses: 'text-gray font-sans text-xxs font-medium',
        priceLevelClasses: 'text-black font-heavy',
      },
    },
  };
};
