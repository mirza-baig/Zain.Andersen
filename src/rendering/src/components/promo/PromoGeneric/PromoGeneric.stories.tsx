// Global
import { Story, Meta } from '@storybook/react';
// Local
import PromoGeneric, { PromoGenericProps } from './PromoGeneric';
import { leftImage, rightImage } from './PromoGeneric.mock-data';

export default {
  title: 'Promo/Promo Generic',
  component: PromoGeneric,
  argTypes: {},
} as Meta;

const Template: Story<PromoGenericProps> = (props) => <PromoGeneric {...props} />;
export const LeftImage = Template.bind({});
LeftImage.args = leftImage;

export const RightImage = Template.bind({});
RightImage.args = rightImage;
