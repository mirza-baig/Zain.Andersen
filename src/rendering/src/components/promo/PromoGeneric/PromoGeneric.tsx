import { Component } from 'src/helpers/Component';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { MediaPrimary } from 'src/helpers/Media';
import { Headline } from 'src/helpers/Headline';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { useTheme } from 'lib/context/ThemeContext';
import { PromoGenericTheme } from './PromoGeneric.theme';
import { Eyebrow } from 'src/helpers/Eyebrow';
import { BodyCopy } from 'src/helpers/BodyCopy';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';
import SingleButton from 'src/helpers/SingleButton/SingleButton';
import PriceLevel from 'src/helpers/PriceLevel/PriceLevel';
import { getEnum } from 'lib/utils';
import { ComponentBackgroundVariants } from 'src/helpers/Component/Component';

export type PromoGenericProps =
  Feature.EnterpriseWeb.Enterprise.Components.Promo.PromoGeneric.PromoGeneric;

const PromoGeneric = (props: PromoGenericProps): JSX.Element => {
  console.log('PromoGeneric', props.fields);
  const { themeData } = useTheme(
    PromoGenericTheme(
      props?.fields?.imgPosition,
      props?.fields?.imageRatio,
      props?.fields?.useLegalCopyFont?.value || false
    )
  );
  const priceLevel = parseInt(props.fields?.priceLevel?.fields.priceLevelText?.value || '');
  const CTA2Props = {
    cta1Link: {
      value: {
        href: props?.fields?.cta2Link?.value?.href,
        text: props?.fields?.cta2Link?.value?.text,
        anchor: props?.fields?.cta2Link?.value?.anchor,
        linktype: props?.fields?.cta2Link?.value?.linktype,
        class: props?.fields?.cta2Link?.value?.class,
        title: props?.fields?.cta2Link?.value?.title,
        target: props?.fields?.cta2Link?.value?.target,
        querystring: props?.fields?.cta2Link?.value?.querystring,
        id: props?.fields?.cta2Link?.value?.id,
      },
    },
    cta1ModalLinkText: {
      value: '',
    },
    cta1AriaLabel: {
      value: '',
    },
    cta1Style: {
      id: props?.fields?.cta2Style?.id,
      url: props?.fields?.cta2Style?.url,
      name: props?.fields?.cta2Style?.name,
      displayName: props?.fields?.cta2Style?.displayName,
      fields: {
        Value: {
          value: props?.fields?.cta2Style?.fields?.Value?.value,
        },
      },
      templateId: props?.fields?.cta2Style?.templateId,
      templateName: props?.fields?.cta2Style?.templateName,
    },
    cta1Icon: {
      id: props?.fields?.cta2Icon?.templateId,
      url: props?.fields?.cta2Icon?.url,
      name: props?.fields?.cta2Icon?.name,
      displayName: props?.fields?.cta2Icon?.displayName,
      fields: {
        Value: {
          value: props?.fields?.cta2Icon?.fields?.Value?.value,
        },
      },
      templateId: props?.fields?.cta2Icon?.templateId,
      templateName: props?.fields?.cta2Icon?.templateName,
    },
  };
  const style = getEnum<ComponentBackgroundVariants>(props.fields?.backgroundColor) ?? 'white';

  return (
    <Component
      variant="full"
      gap="gap-x-0"
      padding="px-0"
      backgroundVariant={style}
      dataComponent="general/promogeneric"
      {...props}
    >
      <div className="col-span-12">
        <div className="py-4 md:max-w-screen-lg lg:mx-auto">
          <div className={themeData.classes.wrapperClass}>
            <div className={themeData.classes.txtDivClass}>
              <div>
                <Eyebrow classes={themeData.classes.eyebrowClass} {...props} />
                <Headline classes={themeData.classes.headlineClass} {...props} />
                {priceLevel ? (
                  <div className={themeData.classes.priceLevelWrapper}>
                    <PriceLevel
                      priceLevel={priceLevel}
                      priceClasses={themeData.classes.priceTextClasses}
                      priceLevelClasses={themeData.classes.priceLevelClasses}
                    />
                  </div>
                ) : (
                  <></>
                )}
                <BodyCopy classes={themeData.classes.bodyClass} {...props} />
                <div className="mb-8">
                  <SingleButton classes={themeData.classes.buttonClasses} {...props} />
                  {props?.fields?.cta2Link?.value?.text && (
                    <SingleButton classes={themeData.classes.buttonClasses} fields={CTA2Props} />
                  )}
                </div>
                <div className="items-bottom">
                  <RichTextWrapper
                    classes={themeData.classes.bottomHeadingClass}
                    field={props.fields?.bottomCaptionHeadline}
                  />
                  <RichTextWrapper
                    classes={themeData.classes.bottomDescriptionClass}
                    field={props.fields?.bottomCaptionDescription}
                  />
                </div>
              </div>
            </div>
            <div className={themeData.classes.imageDivClass}>
              <MediaPrimary {...props} imageLayout="intrinsic" maxH={'h-full'} />
            </div>
          </div>
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<PromoGenericProps>(PromoGeneric);
