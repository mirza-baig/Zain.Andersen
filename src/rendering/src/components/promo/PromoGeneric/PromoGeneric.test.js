// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import PromoGeneric from './PromoGeneric';
import { leftImage, rightImage } from './PromoGeneric.mock-data';

it('leftImage renders correctly', async () => {
  const component = snapshot(PromoGeneric, { componentProps: leftImage });
  await hasDataComponent(component, 'general/promogeneric');
});

it('rightImage renders correctly', async () => {
  const component = snapshot(PromoGeneric, { componentProps: rightImage });
  await hasDataComponent(component, 'general/promogeneric');
});
