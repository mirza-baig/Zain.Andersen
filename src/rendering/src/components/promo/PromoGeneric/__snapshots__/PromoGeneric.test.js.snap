// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leftImage renders correctly 1`] = `
<section
  data-component="general/promogeneric"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-x-0 "
    >
      <div
        class="col-span-12"
      >
        <div
          class="py-4 md:max-w-screen-lg lg:mx-auto"
        >
          <div
            class="grid grid-cols-2 gap-s px-m md:grid-cols-12 md:gap-s promo-generic-wrapper"
          >
            <div
              class="order-last flex items-center pt-2 col-span-6 my-auto md:col-start-8 md:col-span-5 h-full max-w-[480px]"
            >
              <div>
                <div
                  class="text-sm-xxs lg:text-xxs text-dark-gray uppercase font-extralight mb-4"
                >
                  <h3>
                    This is a subheadline
                  </h3>
                </div>
                <div
                  class="items-top font-sans [&_a:hover]:underline text-sm-m lg:text-m font-heavy mb-4"
                >
                  <h2>
                    Generic Promo Headline Text
                  </h2>
                </div>
                <div
                  class="text-body text-dark-gray mb-4"
                >
                  <div
                    class="body-copy"
                  >
                    <div>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit lectus, et sagittis nibh risus vel metus. Quisque eu ornare ante, et gravida mauris.
                    </div>
                  </div>
                </div>
                <div
                  class="mb-8"
                >
                  <div
                    class="mb-m mb-s flex items-start md:flex-row md:items-center"
                  >
                    <a
                      aria-label="Learn More (Opens in a new tab)"
                      class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
                      data-gtm-click=""
                      data-gtm-dl-event="cta_click"
                      href="/"
                      role="link"
                      tabindex="0"
                      target="_blank"
                      title=""
                    >
                      Learn More
                      <span
                        class="ml-xxs"
                      />
                      <span
                        class="sr-only"
                      >
                         (Opens in a new tab)
                      </span>
                    </a>
                  </div>
                </div>
                <div
                  class="items-bottom"
                >
                  <div
                    class="body-copy !font-sans text-sm-xxs lg:text-xxs font-bold mb-2 uppercase"
                  >
                    <div>
                      BOTTOM HEADLINE
                    </div>
                  </div>
                  <div
                    class="body-copy text-body text-dark-gray mb-4"
                  >
                    <div>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit.
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="order-first col-span-6 cols-start-1 md:col-start-1 md:col-span-7 h-full"
            >
              <div
                class="relative hidden md:block  "
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27900%27%20height=%27600%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Unsplash Source"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="relative w-full md:hidden"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27900%27%20height=%27600%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Unsplash Source"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="caption"
              >
                <div
                  class="text-dark-gray mt-xxs text-left mb-s font-regular [&_strong]:text-black  italic font-sans text-sm-xxs md:text-caption"
                >
                  sample caption goes here.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`rightImage renders correctly 1`] = `
<section
  data-component="general/promogeneric"
  id="id1695242607"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-x-0 "
    >
      <div
        class="col-span-12"
      >
        <div
          class="py-4 md:max-w-screen-lg lg:mx-auto"
        >
          <div
            class="grid grid-cols-2 gap-s px-m md:grid-cols-12 md:gap-s promo-generic-wrapper"
          >
            <div
              class="order-last md:order-first flex items-center pt-2 col-span-6 my-auto md:col-start-1 md:col-span-5 h-full max-w-[480px]"
            >
              <div>
                <div
                  class="text-sm-xxs lg:text-xxs text-dark-gray uppercase font-extralight mb-4"
                >
                  <h3>
                    This is a subheadline
                  </h3>
                </div>
                <div
                  class="items-top font-sans [&_a:hover]:underline text-sm-m lg:text-m font-heavy mb-4"
                >
                  <h2>
                    Generic Promo Headline Text
                  </h2>
                </div>
                <div
                  class="text-body text-dark-gray mb-4"
                >
                  <div
                    class="body-copy"
                  >
                    <div>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit lectus, et sagittis nibh risus vel metus. Quisque eu ornare ante, et gravida mauris.
                    </div>
                  </div>
                </div>
                <div
                  class="mb-8"
                >
                  <div
                    class="mb-m mb-s flex items-start md:flex-row md:items-center"
                  >
                    <a
                      aria-label="Learn More (Opens in a new tab)"
                      class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
                      data-gtm-click=""
                      data-gtm-dl-event="cta_click"
                      href="/"
                      rel="noopener noreferrer"
                      role="link"
                      tabindex="0"
                      target="_blank"
                      title=""
                    >
                      Learn More
                      <span
                        class="ml-xxs"
                      />
                      <span
                        class="sr-only"
                      >
                         (Opens in a new tab)
                      </span>
                    </a>
                  </div>
                </div>
                <div
                  class="items-bottom"
                >
                  <div
                    class="body-copy !font-sans text-sm-xxs lg:text-xxs font-bold mb-2 uppercase"
                  >
                    <div>
                      BOTTOM HEADLINE
                    </div>
                  </div>
                  <div
                    class="body-copy text-body text-dark-gray mb-4"
                  >
                    <div>
                      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut semper, justo eget vehicula vestibulum, enim enim suscipit.
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="order-first md:order-last col-span-6 cols-start-1 md:col-start-6 md:col-span-7 h-full"
            >
              <div
                class="relative hidden md:block  "
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27900%27%20height=%27600%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Unsplash Source"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="relative w-full md:hidden"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27900%27%20height=%27600%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Unsplash Source"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="caption"
              >
                <div
                  class="text-dark-gray mt-xxs text-left mb-s font-regular [&_strong]:text-black  italic font-sans text-sm-xxs md:text-caption"
                >
                  sample caption goes here.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
